src/App.tsx

```diff
@@ -20,12 +23,21 @@ export const App: React.FC = () => {
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                value={query}
+                onChange={(event) => {
+                  setQuery(event.target.value);
+                }}
               />
             </div>
           </div>
         </div>
 
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList
+          allMovies={moviesFromServer}
+          visibleMovies={movies}
+          changeMovies={setNewMovies}
```

Comment:
don't pass state function in props, you can create this function (with filtering) in App component and pass this function to the MoviesList